src/js/components/HealthBar.js

```diff
@@ -0,0 +1,68 @@
+import classNames from 'classnames';
+import React from 'react';
+import {Tooltip} from 'reactjs-components';
+
+import HealthBarMap from '../constants/HealthBar';
+import StatusBar from './StatusBar';
+
+class HealthBar extends React.Component {
+  renderToolTip() {
+    let {tasks, instancesCount} = this.props;
+
+    tasks = Object.keys(tasks).filter(function (task) {
+      return tasks[task] !== 0 && task !== 'tasksRunning';
+    }).map(function (task, index) {
+      let percentage = parseInt(tasks[task] / instancesCount * 100, 10);
+
+      let classSet = classNames(HealthBarMap[task].cssName, 'dot icon');
+
+      return (
+        <div key={index}>
+          <span className={classSet}></span>
+          {` ${tasks[task]} ${HealthBarMap[task].label} (${percentage} %)`}
+        </div>
+      );
+    }).filter(function (item) {
+      return item != null;
+    });
+
+    if (tasks.length === 0) {
+      return 'No Running Tasks';
+    }
```

Comment:
I am with you on this one, but in this case it is not possible tasks is not an array there and the decision whether we have something to show or not is decided by the filter in line 12.